File: Tools/TMG_6.0R7/perl/stemmer.pl
#!/usr/bin/perl -w
# stemmer.pl is an interface to the Stemmer module. 
#
# Copyright 2011 Dimitrios Zeimpekis, Eugenia Maria Kontopoulou, Efstratios Gallopoulos

use lib $ARGV[0];
use Stemmer;

$filename=$ARGV[1];
open FD, '<'.$filename;

#$tokens=<FD>;
#@tokens=split(/\s+/, $tokens); 
#print $#tokens+1, "\n";
@tokens=<FD>;

for ($i=0;$i<$#tokens+1;$i++) {
	chomp $tokens[$i];
	$tokens[$i]=Stemmer::stemmer_fun($tokens[$i], 0);
}

$,="\n";
close FD; open FD, '>'.$filename;
print FD @tokens;
close FD; 

File: Tools/TMG_6.0R7/stemmer.m
function s=stemmer(token, dsp)
% STEMMER - applies the Porter's Stemming algorithm [1]
%   S = STEMMER(TOKEN, DSP) returns in S the stemmed word of 
%   TOKEN. DSP indicates if the function displays the result 
%   of each stem (1).
%
%   REFERENCES:
%   [1] M.F.Porter, An algorithm for suffix stripping, Program, 
%   14(3): 130-137, 1980.
%
% Copyright 2011 Dimitrios Zeimpekis, Eugenia Maria Kontopoulou, Efstratios Gallopoulos

error(nargchk(2, 2, nargin));
s=stemmer_p(token, dsp);